src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -75,10 +77,12 @@ type blockWithMeta struct {
 func NewPromReadHandler(
 	engine *executor.Engine,
 	tagOpts models.TagOptions,
-) http.Handler {
+	limitsCfg *config.LimitsConfiguration,
```

Comment:
Can you either do a nil check and return error if this is not set, or just pass it by value?  

Otherwise if someone doesn't set the limits config then we'll get a nil exception when we call validate request while handling a request at runtime I believe?